packages/puppeteer-core/src/bidi/core/UserContext.ts

```diff
@@ -59,11 +63,13 @@ export class UserContext extends EventEmitter<{
   }
 
   #initialize() {
-    // ///////////////////////
+    // ////////////////////
     // Session listeners //
-    // ///////////////////////
-    const session = this.#session;
-    session.on('browsingContext.contextCreated', info => {
+    // ////////////////////
+    const sessionEmitter = this.#disposables.use(
+      new EventEmitter(this.#session)
+    );
+    sessionEmitter.on('browsingContext.contextCreated', info => {
```

Comment:
Let's either use `session` or `sessionEmitter`, using both create more mental overhead.

Reply:
I've settled on `sessionEmitter` due to name clashes